### ISSUE TITLE
Displaying Error / 4.2.7 should fix this problem.

### ISSUE THREAD

**Issue by torstenhoppe** [ttc-freebies/plugin-responsive-images#556]
Sorry for my delay, but it won't work for me. When Content - Responsive Images is activated I can't enter my dashboard in the backend - the Error-Message is: 0 Class "Imagick" not found. And the same issue is frontend, the complete website is not shown.

![Bildschirmfoto 2023-11-09 um 13 03 59](https://user-images.githubusercontent.com/78491005/281729172-41ac2696-0e54-425e-b75d-5fc71cedc77e.png)
![Bildschirmfoto 2023-11-09 um 13 05 35](https://user-images.githubusercontent.com/78491005/281729179-f7ca4f56-4e61-4bdb-b529-62f0d0dfa836.png)

4.2.7 should fix this problem.
If not please reopen this issue

Thanks

_Originally posted by @dgrammatiko in https://github.com/ttc-freebies/plugin-responsive-images/issues/541#issuecomment-1773480134_

**Comment by torstenhoppe** [comment]
Maybe it is a problem with my server configuration. I will check this out next couple days and will report again.

**Comment by torstenhoppe** [comment]
Hi there, just a little late, but it was'n possible earlier. So for me it's not working in version 4.2.7. In the meanwhile I updated my system to Joomla 5.0.1 my webserver is NGINX 1.25.3 with PHP 8.2.12 and MySQL 8.2.0.

For a better understanding I'd like to reply my problem again: In Joomla Backend I created a custom admin modul with a simple HTML table with 2 colums. The first is just coms text and 2nd is a picture. Inside the editor I can define some parameters for this picture, espacialy the width and height.
When I deactivate the plugin the HTML-Code is:
`
<img style="display: block; margin-left: auto; margin-right: auto;" src="../images/path/picture.png" width="300" height="149" loading="lazy" data-path="local-images:/path/picture.png">`
But when I activate the plugin it seems not interpreting this paramaters as expected. The HTML-Code is:
`
<picture class="responsive-image">
                        <source type="image/webp"
                            srcset="/media/cached-resp-images/images/logoicon/hms-media/hms-MEDIA_Logo-color_1920.webp?version=a752a6d1a2722932ccae322632689a02 1920w, /media/cached-resp-images/images/logoicon/hms-media/hms-MEDIA_Logo-color_1600.webp?version=1b956a721050e29470d9c03b813925e8 1600w, /media/cached-resp-images/images/logoicon/hms-media/hms-MEDIA_Logo-color_1200.webp?version=521e895a85b0f8940e4ebdc9d389ec32 1200w, /media/cached-resp-images/images/logoicon/hms-media/hms-MEDIA_Logo-color_992.webp?version=350225c0476c6be2ad0fad9f6fd4d2ed 992w, /media/cached-resp-images/images/logoicon/hms-media/hms-MEDIA_Logo-color_768.webp?version=eadeeccad8bb631c183e87e20cec34c2 768w, /media/cached-resp-images/images/logoicon/hms-media/hms-MEDIA_Logo-color_480.webp?version=4ea116c9fd81938ab10b050f17615c1a 480w, /media/cached-resp-images/images/logoicon/hms-media/hms-MEDIA_Logo-color_320.webp?version=99e60e5b829ec18438fe27f70a86b9a9 320w, /media/cached-resp-images/images/logoicon/hms-media/hms-MEDIA_Logo-color_200.webp?version=c52965b09cc4a1722238de37a7c0f1eb 200w">
                        <source type="image/png"
                            srcset="/media/cached-resp-images/images/logoicon/hms-media/hms-MEDIA_Logo-color_1920.png?version=b17b888372687bc804cecdd0a34f9e0e 1920w, /media/cached-resp-images/images/logoicon/hms-media/hms-MEDIA_Logo-color_1600.png?version=29998e0ffb008fe72083fbbb4f5ffc3f 1600w, /media/cached-resp-images/images/logoicon/hms-media/hms-MEDIA_Logo-color_1200.png?version=626ddda17b12e7b29021e7546d03c6af 1200w, /media/cached-resp-images/images/logoicon/hms-media/hms-MEDIA_Logo-color_992.png?version=44a913d1d6f7336962c71e244fa33f4b 992w, /media/cached-resp-images/images/logoicon/hms-media/hms-MEDIA_Logo-color_768.png?version=ce37dba83e93decd8c6df82a5ae906c1 768w, /media/cached-resp-images/images/logoicon/hms-media/hms-MEDIA_Logo-color_480.png?version=6f5069a5ce8bd9a40c196815f2ba2339 480w, /media/cached-resp-images/images/logoicon/hms-media/hms-MEDIA_Logo-color_320.png?version=74c2414bb2ede8012972cb1284b1559a 320w, /media/cached-resp-images/images/logoicon/hms-media/hms-MEDIA_Logo-color_200.png?version=48fc89198e94861ad3127e2a8c0b950f 200w">
                        <img decoding="async" style="display: block; margin-left: auto; margin-right: auto;"
                            src="/images/logoicon/hms-media/hms-MEDIA_Logo-color.png" width="3448" height="1714"
                            loading="lazy" data-path="local-images:/logoicon/hms-media/hms-MEDIA_Logo-color.png">
                    </picture>`

**Comment by torstenhoppe** [comment]
Hi there, just a little late, but it was'n possible earlier. So for me it's not working in version 4.2.7. In the meanwhile I updated my system to Joomla 5.0.1 my webserver is NGINX 1.25.3 with PHP 8.2.12 and MySQL 8.2.0.

For a better understanding I'd like to reply my problem again: In Joomla Backend I created a custom admin modul with a simple HTML table with 2 colums. The first is just coms text and 2nd is a picture. Inside the editor I can define some parameters for this picture, espacialy the width and height.
When I deactivate the plugin the HTML-Code is:

```
<img style="display: block; margin-left: auto; margin-right: auto;" src="../images/path/picture.png" width="300" height="149" loading="lazy" data-path="local-images:/path/picture.png">
```
![231129-01](https://github.com/ttc-freebies/plugin-responsive-images/assets/78491005/65649fa6-4aba-413a-8148-19a76fdd1d58)


But when I activate the plugin it seems not interpreting this paramaters as expected. The HTML-Code is:

```
<picture class="responsive-image">
<source type="image/webp" srcset="/media/cached-resp-images/path/picture_1920.webp?version=xyz 1920w, /media/cached-resp-images/images/path/picture_1600.webp?version=xyz 1600w, /media/cached-resp-images/images/path/picture_1200.webp?version=xyz >

<source type="image/png" >
<img decoding="async" style="display: block; margin-left: auto; margin-right: auto;"
src="/images/path/picture.png" width="3448" height="1714"
loading="lazy" data-path="local-images:/path/picture.png">
</picture>
```

And here it takes the original width and height of the image and doesn't interpret the parameter that I captured in the form (see screenshot)

**Comment by dgrammatiko** [comment]
>  espacialy the width and height

The width and height are automatically extracted from the original image. The reason is that these values are there so the browser could calculate the pre-rendered area and not to set the size of the image. The whole idea with responsive images is that you set a container size and then the image will render inside it with a width 100% and height auto. So, in your case wrap the image in a div with a class (or inline style) to set the preferred size


About the size part: the calculated aspect ratio in the 1st case is width/height = 300/149 = 2.013
in the case of responsive images is width/height = 3448/1714 = 2.011
So, essentially the aspect ratio is the same

**Comment by torstenhoppe** [comment]
Yes I understand, but if I have adjusted the values within the editor, then I expect them to be applied and not recalculate the original values from the size of the file, right?
The problem is still that I get a much too large display of the image - which I did not expect.

**Comment by dgrammatiko** [comment]
> then I expect them to be applied and not recalculate the original values from the size of the file, right?

Nope, the size are always reflect the given image size, that's by design

> then I expect them to be applied and not recalculate the original values from the size of the file, right?

Could you send me the generated  html for the module at d.grammatiko at gmail dot com?

**Comment by dgrammatiko** [comment]
Is this still a problem?

I'm closing this issue, if there's still a problem please reopen the issue or create a new one
